devicetypes/stelpro/stelpro-ki-zigbee-thermostat.src/stelpro-ki-zigbee-thermostat.groovy

```diff
@@ -310,13 +313,16 @@ def parse(String description) {
 				def intVal = Integer.parseInt(descMap.value, 16)
 				log.debug "HEAT DEMAND"
 				map.name = "thermostatOperatingState"
-				if (intVal < 10) {
+				if (intVal < 0x10) {
 					map.value = "idle"
 				} else {
 					map.value = "heating"
 				}
 
-				if (settings.heatdetails == "No") {
+				// If the user does not want to see the Idle and Heating events in the event history,
+				// don't show them. Otherwise, don't show them more frequently than 30 seconds.
+				if (settings.heatdetails == "No" ||
+					!secondsPast(device.currentState("thermostatOperatingState").getLastUpdated(), 30)) {
```

Comment:
I did this to try to compensate for the change to the operating state reporting intervals in configure(). I don't have a problem removing the extra 30 second logic, though.

cc @stelpro

Reply:
This is good idea.
The 30 seconds could be set to another value, but I suggest to keep it that way for now.